PROTOCOL.md

```diff
@@ -0,0 +1,99 @@
+# GraphQL over WebSocket Protocol
+
+## Client-server communication
+
+Each message has a `type` field, which defined in the protocol of this package, as well as associated fields inside `payload` field, depending on the message type, and `id` field so the client can identify each response from the server.
+
+The structure of each client side message is:
+```typescript
+export interface OperationMessage {
+  payload?: any;
+  id?: string;
+  type: string;
+}
+```
+
+### Client -> Server
+
+#### GQL_CONNECTION_INIT
+Client sends this message after plain websocket connection to start the communication with the server
+
+This message also triggers the `onConnect` callback on the server.
```

Comment:
Im not sure this is the right place for such comment,
maybe we need a seperate documentation about events,
but this is Implementation specific and not Protocl specific info.

This ofcourse applies to the rest of the document as well.